### PR TITLE
Decompile bag.s

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: arm9/asm/scrcmd_18.s [hunk 1]
```diff
@@ -615,8 +618,8 @@ _020447F2:
 	pop {r4-r6, pc}
 	.balign 4
 
-	thumb_func_start FUN_02044800
-FUN_02044800: ; 0x02044800
+	thumb_func_start ScrCmd_findpartymonwithnature
```
Comment (red031000) [inline arm9/asm/scrcmd_18.s:621]: possibly hard to distinguish words here?

File: arm9/src/bag.c [hunk 2]
```diff
@@ -0,0 +1,463 @@
...
+{
+    MI_CpuClear16(bag, sizeof(struct Bag));
+}
+
+void Sav2_Bag_copy(const struct Bag * src, struct Bag * dest)
+{
+    MI_CpuCopy8(src, dest, sizeof(struct Bag));
+}
+
+u32 Bag_GetRegisteredItem(struct Bag * bag)
```
Comment (red031000) [inline arm9/src/bag.c:31]: inconsistent naming

File: arm9/src/main.c [hunk 3]
```diff
@@ -73,7 +73,7 @@ THUMB_FUNC void NitroMain(void)
     FUN_02002C50(1, 3);
     FUN_02002C50(3, 3);
     gBacklightTop.unk18 = -1;
-    gBacklightTop.unk20 = FUN_0202254C();
+    gBacklightTop.unk20 = SaveBlock2_new();
```
Comment (red031000) [inline arm9/src/main.c:76]: I'm unsure if this should even be called gBacklightTop anymore, it clearly contains a lot more i.e. save block 2

File: arm9/asm/scrcmd_18.s (hunk 1 above)
Comment (PikalaxALT) [reply arm9/asm/scrcmd_18.s:621]: crameverythingtogetherlowercase is the standard for script commands

File: arm9/src/main.c (hunk 3 above)
Comment (PikalaxALT) [reply arm9/src/main.c:76]: Imma leave it until we decide for sure what to call it.

File: arm9/src/bag.c (hunk 2 above)
Comment (PikalaxALT) [reply arm9/src/bag.c:31]: Elaborate please?

Comment (red031000) [reply arm9/src/bag.c:31]: some funcs are called Bag_ and some are Bag without the underscore

File: arm9/asm/scrcmd_18.s (hunk 1 above)
Comment (red031000) [reply arm9/asm/scrcmd_18.s:621]: fair enough, I can just see it being a bit difficult to separate words

File: arm9/src/main.c (hunk 3 above)
Comment (red031000) [reply arm9/src/main.c:76]: ok

**Pull request merged**